src/main/java/org/opensearch/neuralsearch/sparse/query/SparseQueryWeight.java

```diff
@@ -148,6 +155,33 @@ private Scorer selectScorer(SparseVectorQuery query, LeafReaderContext context,
         );
     }
 
+    private SparseVectorReader getSparseVectorReader(LeafReader leafReader, String fieldName) throws IOException {
+        BinaryDocValues docValues = leafReader.getBinaryDocValues(fieldName);
+        if (docValues instanceof SparseBinaryDocValuesPassThrough sparseBinaryDocValuesPassThrough) {
+            SparseVectorReader reader;
+            SegmentInfo segmentInfo = sparseBinaryDocValuesPassThrough.getSegmentInfo();
+            InMemoryKey.IndexKey key = new InMemoryKey.IndexKey(segmentInfo, fieldName);
+            SparseVectorForwardIndex index = InMemorySparseVectorForwardIndex.get(key);
+            if (index != null) {
+                SparseVectorReader inMemoryReader = index.getReader();
+                SparseVectorWriter inMemoryWriter = index.getWriter();
+                reader = (docId) -> {
+                    SparseVector vector = inMemoryReader.read(docId);
+                    if (vector != null) {
+                        return vector;
+                    }
+                    vector = sparseBinaryDocValuesPassThrough.read(docId);
```

Comment:
I'd rather put your cache specific logic into your cache related class instead of spreading it in other places.